drivers/platform/maxim/gpio_extra.h

```diff
@@ -0,0 +1,79 @@
+/***************************************************************************//**
+ *   @file   maxim/gpio_extra.h
+ *   @brief  Header file for maxim gpio specifics.
+ *   @author Ciprian Regus (ciprian.regus@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef GPIO_EXTRA_H_
+#define GPIO_EXTRA_H_
+
+#include "stdbool.h"
+#include <stdint.h>
+#include "no-os/irq.h"
+#include "no-os/gpio.h"
+
+#define N_PINS	14
```

Comment:
So I see that you've included these here, but I think you can drop the N_PINS and N_PORTS checks because the Maxim HAL will check these. If there is a bug in Maxim HAL and it doesn't check them, and it crashes the program, then it's fine to add the checks ourselves, but I'm pretty sure that GPIO_Config() will return some meaningful error code if you pass it port -1 and pin 1234. So you simply need to translate that error code into no-os error codes and pass it back.

Reply:
I checked that, and GPIO_Config() doesn't check for either the port or pin number (passing an invalid port results in a null pointer being dereferenced). So, in this case, I'll keep those defines.

Reply:
ok that's fine, it seems they have an issue in their HAL so it's fine to keep the checks, although you might have a problem then trying to keep up with all devices' available pins, you'll have to do something like:
```
#ifdef __MAX12345
#define N_PINS 14
#elif defined(__MAX54321)
#define N_PINS 16
...
```